catalog/core/catalog-core-localstorageprovider/src/main/java/org/codice/ddf/catalog/content/impl/FileSystemStorageProvider.java

```diff
@@ -87,6 +90,8 @@
 
     private Map<String, List<Metacard>> deletionMap = new ConcurrentHashMap<>();
 
+    private Map<String, List<String>> updateMap = new ConcurrentHashMap<>();
```

Comment:
I would suggest looking at Guava's [`Multimap`](http://google.github.io/guava/releases/19.0/api/docs/com/google/common/collect/Multimap.html). It's a lot cleaner for things like this.